### PR TITLE
Concurrency issues, bug fix.

### PR DESCRIPTION
**Pull request by AOzmond**
Added mutex to Linux device name caching to avoid concurrent map issues.

### CODE REVIEW

File: lib/devicename_linux.go [hunk 1]
```diff
@@ -57,6 +69,8 @@ func enumerateAndCache() {
 
 	devices, _ := e.Devices()
 
+	newCache := make(map[string]deviceInfo)
```
Comment (lietu) [inline lib/devicename_linux.go]: Why not just
```suggestion
	var newCache map[string]deviceInfo{}
```
?

File: lib/devicename_linux.go [hunk 2]
```diff
@@ -14,7 +15,10 @@ type deviceInfo struct {
 	Speed string
 }
 
-var deviceInfoCache = map[string]deviceInfo{}
+var (
+	deviceInfoCache   = map[string]deviceInfo{}
+	deviceInfoCacheMu sync.RWMutex
```
Comment (lietu) [inline lib/devicename_linux.go]:
```suggestion
	deviceInfoCacheLock sync.RWMutex
```

File: lib/devicename_linux_test.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+	var wg sync.WaitGroup
+	wg.Add(goroutines * 2)
+
+	// Readers
+	for i := 0; i < goroutines; i++ {
+		go func() {
+			defer wg.Done()
+			for j := 0; j < iterations; j++ {
+				for _, dev := range testDevices {
```
Comment (lietu) [inline lib/devicename_linux_test.go:34]:
```suggestion
			for j := 0; j < iterations; j++ {
			    // Add random sleep between 0 and 3ms here
				for _, dev := range testDevices {
```

File: lib/devicename_linux_test.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+			}
+		}()
+	}
+
+	// Writers (clear cache)
+	for i := 0; i < goroutines; i++ {
+		go func() {
+			defer wg.Done()
+			for j := 0; j < iterations; j++ {
+				for _, dev := range testDevices {
```
Comment (lietu) [inline lib/devicename_linux_test.go]:
```suggestion
			for j := 0; j < iterations; j++ {
			    // Add random sleep between 0 and 3ms here			
				for _, dev := range testDevices {
```

**Review by lietu — CHANGES_REQUESTED** [review]
Just leave the first test case, you're dedicating entirely too much code to testing a false-positive edge case from an automated tool. Added a few suggestions.

**Review by lietu — APPROVED** [review]
